### PR TITLE
feat: Add custom training integration with experiments notebook

### PR DESCRIPTION
**Pull request by inardini**
<br>
This notebook shows how to autolog paramenters and metrics of an ML experiment running on Vertex AI training by leveraging the integration with Vertex AI Experiments.
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb [hunk 1]
```diff
@@ -0,0 +1,884 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>heading sentence case: Local -&gt; local</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>Vertex AI training -&gt; Vertex AI Training</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>replace -q with --quiet</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>set IS_COLAB to False</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>Point out the parameter enable_autolog</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>its not clear what this code does. Add a text description.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    if delete_bucket or os.getenv("IS_TESTING"):
</pre><p>I think you meant delete_experiment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1887/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/get_started_with_custom_training_autologging_local_script.ipynb:1]: <p>done</p>

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**